### DISCUSSION TITLE
This job is awaiting completion.

### DISCUSSION THREAD

**Discussion by bayden10** [Security-Onion-Solutions/securityonion#2233] [Unsupported Versions] [answered]
Hi,

SO v2.3.10 Install from ISO
Standalone Deployment
Host 
Hyper-V VM
CPU x10
Mem 32GB
Storage 217GB (Fixed not Dynamic)
In 24hrs 1,998,529 logs processed

``Checking Docker status

    Docker -------------------------------------------------------------------------------------------- [ OK ]

Checking container statuses

    so-aptcacherng ------------------------------------------------------------------------------------ [ OK ]
    so-cortex ----------------------------------------------------------------------------------------- [ OK ]
    so-curator ---------------------------------------------------------------------------------------- [ OK ]
    so-dockerregistry --------------------------------------------------------------------------------- [ OK ]
    so-elastalert ------------------------------------------------------------------------------------- [ OK ]
    so-elasticsearch ---------------------------------------------------------------------------------- [ OK ]
    so-filebeat --------------------------------------------------------------------------------------- [ OK ]
    so-fleet ------------------------------------------------------------------------------------------ [ OK ]
    so-grafana ---------------------------------------------------------------------------------------- [ OK ]
    so-idstools --------------------------------------------------------------------------------------- [ OK ]
    so-influxdb --------------------------------------------------------------------------------------- [ OK ]
    so-kibana ----------------------------------------------------------------------------------------- [ OK ]
    so-kratos ----------------------------------------------------------------------------------------- [ OK ]
    so-logstash --------------------------------------------------------------------------------------- [ OK ]
    so-mysql ------------------------------------------------------------------------------------------ [ OK ]
    so-nginx ------------------------------------------------------------------------------------------ [ OK ]
    so-playbook --------------------------------------------------------------------------------------- [ OK ]
    so-redis ------------------------------------------------------------------------------------------ [ OK ]
    so-sensoroni -------------------------------------------------------------------------------------- [ OK ]
    so-soc -------------------------------------------------------------------------------------------- [ OK ]
    so-soctopus --------------------------------------------------------------------------------------- [ OK ]
    so-steno ------------------------------------------------------------------------------------------ [ OK ]
    so-strelka-backend -------------------------------------------------------------------------------- [ OK ]
    so-strelka-coordinator ---------------------------------------------------------------------------- [ OK ]
    so-strelka-filestream ----------------------------------------------------------------------------- [ OK ]
    so-strelka-frontend ------------------------------------------------------------------------------- [ OK ]
    so-strelka-gatekeeper ----------------------------------------------------------------------------- [ OK ]
    so-strelka-manager -------------------------------------------------------------------------------- [ OK ]
    so-suricata --------------------------------------------------------------------------------------- [ OK ]
    so-telegraf --------------------------------------------------------------------------------------- [ OK ]
    so-thehive ---------------------------------------------------------------------------------------- [ OK ]
    so-thehive-es ------------------------------------------------------------------------------------- [ OK ]
    so-wazuh ------------------------------------------------------------------------------------------ [ OK ]
    so-zeek ------------------------------------------------------------------------------------------- [ OK ]```

![pending](https://user-images.githubusercontent.com/22577485/101260906-2cbcd580-3701-11eb-844a-7bc69e4fbdca.png)

When trying view PCAPs or to follow _id, states "This job is awaiting completion" on all activities.

![awaiting](https://user-images.githubusercontent.com/22577485/101260974-99d06b00-3701-11eb-81be-8597c8fabb0c.png)

Stenographer seems to be working as expected but perhaps I'm looking in the wrong place.

```ps -aux | grep stenogr+```

```
stenogr+   2899  0.0  0.0  11688  1504 ?        Ss   16:31   0:00 -bash -c /usr/bin/stenographer -syslog=false >> /var/log/stenographer/stenographer.log 2>&1
stenogr+   2915  0.2  0.1 1091904 33908 ?       Sl   16:31   0:04 /usr/bin/stenographer -syslog=false
stenogr+   2936  0.1  6.5 2362848 2102440 ?     Sl   16:31   0:02 /usr/bin/stenotype -v --uid=stenographer --gid=stenographer --threads=1 --dir=/tmp/stenographer034797979 --iface=bond0
root      43822  0.0  0.0 112812   968 pts/0    S+   16:59   0:00 grep --color=auto stenogr+
```

I see stenogr+ writing to the log

```
tail -f /opt/so/log/stenographer/stenographer.log -n100

2020-12-05T17:03:02.590265Z T:b3b307 [index.cc:269] Wrote all index files for /tmp/stenographer034797979/IDX0/.1607187721001567, moving to /tmp/stenographer034797979/IDX0/1607187721001567
2020-12-05T17:03:02.593018Z T:b43317 [aio.cc:122] Closing /tmp/stenographer034797979/PKT0/.1607187721001567 (4), truncating to 40MB and moving to /tmp/stenographer034797979/PKT0/1607187721001567
2020-12-05T17:03:59.053820Z T:b43317 [stenotype.cc:537] Thread 0 stats: MB=1266 secs=1961.9 MBps=0.645292 packets=1533716 blocks=1266 polls=3894 drops=0 drop%=0
2020-12-05T17:04:03.689389Z T:b43317 [aio.cc:190] Opening packet file /tmp/stenographer034797979/PKT0/.1607187843689187: 4
2020-12-05T17:04:03.696830Z T:b3b307 [index.cc:269] Wrote all index files for /tmp/stenographer034797979/IDX0/.1607187782580494, moving to /tmp/stenographer034797979/IDX0/1607187782580494
2020-12-05T17:04:03.709740Z T:b43317 [aio.cc:122] Closing /tmp/stenographer034797979/PKT0/.1607187782580494 (5), truncating to 28MB and moving to /tmp/stenographer034797979/PKT0/1607187782580494
```

I've also restarted Stenographer and same result.

Default is used for `/opt/so/conf/steno/config. PCAPs are present and so are the indexes.

```
{
  "Threads": [
    { "PacketsDirectory": "/nsm/pcap"
    , "IndexDirectory": "/nsm/pcapindex"
    , "MaxDirectoryFiles": 30000
    , "DiskFreePercentage": 10
    }
  ]
  , "StenotypePath": "/usr/bin/stenotype"
  , "Interface": "bond0"
  , "Port": 1234
  , "Host": "127.0.0.1"
  , "Flags": ["-v", "--uid=stenographer", "--gid=stenographer"]
  , "CertPath": "/etc/stenographer/certs"
}
```
Also attempted to import PCAPs from sample so-import from here: `wget https://www.malware-traffic-analysis.net/2020/09/16/2020-09-16-Qakbot-infection-traffic.pcap.zip`

`so-import-pcap 2020-09-16-Qakbot-infection-traffic.pcap`

Getting the same result.

Thx
Jim

**Comment by jertel** [comment, marked as answer]
Take a look at the /opt/so/log/sensoroni/sensoroni.log. Any oddities in there?

**Reply by bayden10** [reply]
Hi Jason,

Thank you for your reply. Hmm yes I see connection refused on port 443 from the external interface facing on the SO host. I was not aware of this hairpin approach I thought the API would connect to the internal IP or localhost?

```
timestamp=2020-12-07T21:42:52.795734758Z level=warn message="Failed to poll for pending jobs" error="Post \"https://soict/sensoroniagents/api/sensor\": dial tcp 42.42.47.48:443: connect: connection refused"
timestamp=2020-12-07T21:42:53.809565592Z level=warn message="Failed to submit request" error="Post \"https://soict/sensoroniagents/api/sensor\": dial tcp 42.42.47.48:443: connect: connection refused"
timestamp=2020-12-07T21:42:53.809644997Z level=warn message="Failed to poll for pending jobs" error="Post \"https://soict/sensoroniagents/api/sensor\": dial tcp 42.42.47.48:443: connect: connection refused"
timestamp=2020-12-07T21:42:54.825568754Z level=warn message="Failed to submit request" error="Post \"https://soict/sensoroniagents/api/sensor\": dial tcp 42.42.47.48:443: connect: connection refused"
timestamp=2020-12-07T21:42:54.825674561Z level=warn message="Failed to poll for pending jobs" error="Post \"https://soict/sensoroniagents/api/sensor\": dial tcp 42.42.47.48:443: connect: connection refused"
```
I reconfigured the SRC-NAT on the core router for this host and it's processing all the jobs now.

Thx
Jim

**Reply by jertel** [reply]
The manager hostname you provide/confirm during setup (one of the first couple of prompts I believe) is asking for the internal hostname of the manager. You appear to have chosen `soict` but your local DNS server is returning a public IP for that hostname instead of the internal IP. You can add an entry in your hosts file if you want to get around this without changing your LAN, to force `soict` to resolve to the internal IP, or 127.0.0.1 since you're running this as a standalone grid.

**Reply by bayden10** [reply]
Yes host name soict was already set in hosts at 127.0.0.1 after install that's why I was a bit baffled. I did add the local IP after noticing that earlier and then adjusted the SRC-NAT and things started to work.

**Reply by cowbe0x004** [reply]
> 
> 
> Yes host name soict was already set in hosts at 127.0.0.1 after install that's why I was a bit baffled. I did add the local IP after noticing that earlier and then adjusted the SRC-NAT and things started to work.

I'm having the same connection refused error from sensoroni when I change my hostname's record to public IP, so I will need to mess with NAT. Could you explain what you did?
Also having an internal ip in /etc/hosts doesn't change this behavior, sensoroni still pick whatever is in the DNS zone.
